zerver/migrations/0595_add_realmexport_table_and_backfill.py

```diff
@@ -0,0 +1,97 @@
+# Generated by Django 5.0.9 on 2024-10-03 07:17
+
+from datetime import datetime, timezone
+
+import django.db.models.deletion
+from django.conf import settings
+from django.db import migrations, models
+from django.db.backends.base.schema import BaseDatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+
+
+def timestamp_to_datetime(timestamp: float | None) -> datetime | None:
+    if timestamp is None:
+        return None
+    return datetime.fromtimestamp(float(timestamp), tz=timezone.utc)
+
+
+def datetime_to_timestamp(dt: datetime) -> int:
+    return int(dt.timestamp())
+
+
+def backfill_realm_export(apps: StateApps, schema_editor: BaseDatabaseSchemaEditor) -> None:
+    REALM_EXPORTED = 206
+    EXPORT_PUBLIC = 1
+    EXPORT_FULL_WITH_CONSENT = 2
+
+    RealmAuditLog = apps.get_model("zerver", "RealmAuditLog")
+    RealmExport = apps.get_model("zerver", "RealmExport")
+
+    for audit_log in RealmAuditLog.objects.filter(event_type=REALM_EXPORTED).order_by("id"):
+        extra_data = audit_log.extra_data
+        started_timestamp = extra_data.get("started_timestamp")
+        failed_timestamp = extra_data.get("failed_timestamp")
+        deleted_timestamp = extra_data.get("deleted_timestamp")
+        export_path = extra_data.get("export_path")
+
+        RealmExport.objects.create(
+            realm=audit_log.realm,
+            acting_user=audit_log.acting_user,
+            # We don't have a way to determine if it was a EXPORT_FULL_WITHOUT_CONSENT.
+            # This is the best we can do.
+            type=EXPORT_PUBLIC if audit_log.acting_user else EXPORT_FULL_WITH_CONSENT,
+            date_requested=audit_log.event_time,
+            date_started=timestamp_to_datetime(started_timestamp),
+            date_failed=timestamp_to_datetime(failed_timestamp),
+            date_deleted=timestamp_to_datetime(deleted_timestamp),
+            export_path=export_path,
```

Comment:
This doesn't set an initial value of `status`. I think probably we can compute it from the `date_` fields...